### PR TITLE
[DSCP-390] Mock time

### PR DESCRIPTION
**Pull request by Martoon-00**
### Description

For the sake of #328 get passing tests in presence of slot verification I had to mock slotting.
However, mocking the time entirely occurred to be even simpler in some sense.

### YT issue

https://issues.serokell.io/issue/DSCP-390

### Checklist

Hint: a perfect PR has all the checkmarks set.

Possible related changes (conditional):
- [x] I added tests if required, in case they are:
  - Covering new introduced functionality (feature).
  - Regression tests preventing the bug from silently reappearing again (bugfix).
- [x] I have checked the [sample config](../tree/master/docs/config-full-sample.yaml) and [launch scripts](../tree/master/scripts/launch) and updated them if it was necessary (especially if executables or CLIs were changed).
- [x] I have checked READMEs and changed them accordingly if it's necessary (the main [README.md](../tree/master/README.md) as well as subproject READMEs for all related executables).
- [x] If a public API structure or/and data serialization was changed, I made sure that these changes are reflected in the:
  - [Swagger](../tree/master/specs/disciplina) API specs.
  - Any [related documentation](../tree/master/docs/api-types.md).
- [x] If any public documentation was written, I have spellchecked it.

Stylistic (obligatory):
- [x] My commit history is clean, descriptive and do not contain merge or revert commits or commits like "WIP".
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).
- [x] My code is documented (haddock) and these docs are decent (full enough and spellchecked).

### CODE REVIEW

File: witness/tests/Test/Dscp/Witness/Explorer/ExplorerSpec.hs [hunk 1]
```diff
@@ -51,12 +53,16 @@ createAndSubmitPub genSecret = do
...
-    => SlotId -> m HeaderHash
-dumpBlock slotId = do
-    block <- createBlock runSdM slotId
-    void $ applyBlock block
-    return (headerHash block)
+    :: (TestWitnessWorkMode ctx m, WithinWriteSDLock)
+    => m HeaderHash
+dumpBlock = do
+    slotId <- rewindToNextSlot
+    let issuerKey = KeyResources . mkSecretKeyData $ testFindSlotOwner slotId
```
Comment (Heimdell) [inline witness/tests/Test/Dscp/Witness/Explorer/ExplorerSpec.hs]: This operation is a bit uncontrollable. I don't think anything can happen so we might want to cancel this, though.

Comment (Martoon-00) [reply witness/tests/Test/Dscp/Witness/Explorer/ExplorerSpec.hs]: Well, that's true. I'd like to come up with a better thing here, but with interface provided by our `witness <-> slot` mapping logic it's probably the only normal way.

Comment (Heimdell) [reply witness/tests/Test/Dscp/Witness/Explorer/ExplorerSpec.hs]: No, I'm about `createBlock` after `sleep`.

Comment (Martoon-00) [reply witness/tests/Test/Dscp/Witness/Explorer/ExplorerSpec.hs]: Lool, that wasn't obvious for me.
Okay, I'll just probably tell in the comment that it's part of `rewindToNextSlot`'s semantics to return _new_ slot. If we cannot guarantee this simple invariant for `rewindToNextSlot`, then the whole mocking doesn't look sound.

**Pull request merged**